homeassistant/components/template/light.py

```diff
@@ -267,38 +347,14 @@ def template_light_startup(event):
             self.async_schedule_update_ha_state()
 
     async def async_update(self):
-        """Update the state from the template."""
-        if self._template is not None:
-            try:
-                state = self._template.async_render().lower()
-            except TemplateError as ex:
-                _LOGGER.error(ex)
-                self._state = None
+        """Update from templates."""
+        await self.update_state()
```

Comment:
Instead of making these awaitable, you can mark them as `@callback` which means you don't need to yield.

Reply:
So, I canmake `update_state()` not async  also?
```
 @callback
  def update_state(self):
        """Update the state from the template."""
        if self._template is not None:
          ...
``

Reply:
I don't get the magic behind, but indeed it works well. Thx.